### PR TITLE
updating readme to be accurate for api endpoints 

### LINKED ISSUE #16: readme does not contain up-to-date information about the endpoints
The endpoints listed in the README.md file are not all entirely accurate. 
It should also include information about setting up the project and running it.

### PR DESCRIPTION
**Pull request by ivoinestrachan**
this closes #16

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,11 +4,41 @@ The SineRider externally available API
 
 ## How to use (endpoints)
 
-- Send a `GET` request to `/all` to get all the scores that have been saved. The scores have been sorted by `charCount` and `T` both in ascending order.
-
-- Send a `GET` request to `/level/<LEVEL_NAME>` (e.g `/level/HELLO_WORLD`) to get all scores for that specific level sorted by `charCount` and `T`.
-
 - Send a `GET` request to `/generate` to generate a new level and store it in airtable. Does not return the URL to the generated game
 
 - Send a `GET` request to `/daily` to get a random level from the airtable to play. Once the level is returned, the game url will be marked as *played* and may not be returned in the future.
```
Comment (JosiasAurel) [inline README.md]: there is no `/daily` endpoint in anymore

File: README.md [hunk 2]
```diff
@@ -4,11 +4,41 @@ The SineRider externally available API
 
 ## How to use (endpoints)
 
-- Send a `GET` request to `/all` to get all the scores that have been saved. The scores have been sorted by `charCount` and `T` both in ascending order.
-
-- Send a `GET` request to `/level/<LEVEL_NAME>` (e.g `/level/HELLO_WORLD`) to get all scores for that specific level sorted by `charCount` and `T`.
-
 - Send a `GET` request to `/generate` to generate a new level and store it in airtable. Does not return the URL to the generated game
```
Comment (JosiasAurel) [inline README.md]: this endpoint now returns the id of the level saved in the airtable, please mention that here

File: README.md [hunk 3]
```diff
@@ -4,11 +4,41 @@ The SineRider externally available API
...
+
+- Send a `GET /reddit/nick` Redirect to a Reddit post for a specific puzzle by its nickname.
+
+### Authenticated Endpoints
+
+_Note: This endpoint require basic authentication._
+
+- Send a `POST /publishNewDailyPuzzle` Publish a new daily puzzle across all bot services. Selects a puzzle from unplayed puzzles and marks it as active.
+
+## Usage
```
Comment (JosiasAurel) [inline README.md]: the setup step should also include the environment variables. 

something like 

```
SINERIDER_TWITTER_URL=<example>
...
```
for all the other variables, don't actually put the environment variables here

**Review by grymmy — APPROVED** [review]
Didn't look too closely at this but will take it on faith this is accurate

**Comment by ivoinestrachan** [comment]
thank you sir 🙏🏾

**Comment by JosiasAurel** [comment]
please refer to [hackclub/scrappy](https://github.com/hackclub/scrappy) for an example readme file

**Review by JosiasAurel — CHANGES_REQUESTED** [review]

File: README.md (hunk 3 above)
Comment (ivoinestrachan) [reply README.md]: let me do this right now

File: README.md [hunk 4]
```diff
@@ -4,11 +4,41 @@ The SineRider externally available API
...
-
 - Send a `GET` request to `/generate` to generate a new level and store it in airtable. Does not return the URL to the generated game
 
 - Send a `GET` request to `/daily` to get a random level from the airtable to play. Once the level is returned, the game url will be marked as *played* and may not be returned in the future.
 
+- Send a `GET /level/name/highscoreType` Retrieve scores for a specific level and score type. Parameters:
+  - `:name`: Name of the level.
+  - `:highscoreType`: Type of highscore to retrieve.
+
+  (e.g `/level/HELLO_WORLD`)
```
Comment (JosiasAurel) [inline README.md]: i think the endpoint has to be of the form `/level/HELLO_WORLD/charCount` i.e you have to specify the high-score type you are looking for

File: README.md [hunk 5]
```diff
@@ -4,11 +4,41 @@ The SineRider externally available API
...
+
+- Send a `GET /levels` Get a list of all available levels, sorted alphabetically.
+
+- Send a `GET /puzzle/nick` Redirect to a specific puzzle by its nickname.
```
Comment (JosiasAurel) [inline README.md]: please use the `:nick` to make it more obvious that this is variable. 
you could also just mention that 'nick' is just a placeholder

File: README.md (hunk 3 above)
Comment (JosiasAurel) [reply README.md]: thank you

**Review by JosiasAurel — CHANGES_REQUESTED** [review]

File: README.md [hunk 6]
```diff
@@ -4,11 +4,64 @@ The SineRider externally available API
...
+  `SINERIDER_AIRTABLE_BASE=<exampele>`
+
+  `SINERIDER_API_SECRET=<exampele>`
+
+  `SINERIDER_REDDIT_API_KEY=<exampele>`
+
+  `SINERIDER_REDDIT_BOT_URL=<exampele>`
+
+  `SINERIDER_TWITTER_API_KEY=<exampele>`
+
```
Comment (JosiasAurel) [inline README.md]: after thought it might be better to remove the ´\<exampele\>` here. could you also copy-pasta this into a .env.example file?

**Review by JosiasAurel — CHANGES_REQUESTED** [review]

**Review by JosiasAurel — APPROVED** [review]
thank you so much @ivoinestrachan

**Pull request merged**